internal/worker/worker.go

```diff
@@ -85,9 +85,13 @@ func (w *BaseWorker) Start(ctx context.Context) error {
 	go func() {
 		for {
 			data, err := w.ioHandler.Read(ctx)
-			if err != nil {
-				readErrChan <- err
-				return
+			if err != nil || len(data) == 0 {
+				select {
+				case readErrChan <- err:
+					return
+				case <-time.After(time.Second):
```

Comment:
Do you think we can use `case <-ctx.Done()` instead?
That would make the behaviour more predictable behaviour.

Reply:
Thanks for the suggestion @soumya-codes to improve coordination. I have made these changes.